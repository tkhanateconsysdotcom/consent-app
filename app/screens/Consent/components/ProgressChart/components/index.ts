import Legend from "./Legend";
import Chart from "./Chart";
import PeriodConsent from "./PeriodConsent";

export { Chart, Legend, PeriodConsent };
