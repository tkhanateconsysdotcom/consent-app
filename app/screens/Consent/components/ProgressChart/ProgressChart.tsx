import React from "react";
import moment from "moment";
import "moment/min/locales";
import { View } from "react-native";

import { Text } from "components";
import { withLocalization, LocalizationContextProps } from "utils";

import styles from "./ProgressChart.styles";
import { Legend, Chart, PeriodConsent } from "./components";

interface Props {
  isMonth?: boolean;
  totalPartners: number;
  foodPartners: number;
  transportPartners: number;
  otherPartners: number;
  monthlyPartnersConsent: number;
}

const ProgressChart = ({
  totalPartners = 0,
  foodPartners = 0,
  transportPartners = 0,
  otherPartners = 0,
  monthlyPartnersConsent = 0,
  isMonth = false,
  language = "",
}: Props & LocalizationContextProps) => {
  if (!monthlyPartnersConsent) {
    return null;
  }

  const periodPartnersConsent = isMonth
    ? monthlyPartnersConsent
    : monthlyPartnersConsent * 12;

  const totalPartnersPercentage =
    totalPartners / periodPartnersConsent > 1
      ? 1
      : totalPartners / periodPartnersConsent;
  const transportPartnersPercentage =
    transportPartners / periodPartnersConsent > 1
      ? 1
      : transportPartners / periodPartnersConsent;
  const foodPartnersPercentage =
    foodPartners / periodPartnersConsent > 1
      ? 1
      : foodPartners / periodPartnersConsent;

  const period = moment()
    .locale(language)
    .format(isMonth ? "MMMM" : "YYYY");

  return (
    <View style={styles.container}>
      <View style={styles.periodContainer}>
        <Text.H3 style={styles.header}>{period}</Text.H3>
      </View>
      <Chart
        totalPartnersPercentage={totalPartnersPercentage}
        transportPartnersPercentage={transportPartnersPercentage}
        foodPartnersPercentage={foodPartnersPercentage}
      />
      <Legend
        totalPartners={totalPartners}
        foodPartners={foodPartners}
        transportPartners={transportPartners}
        otherPartners={otherPartners}
      />
      <PeriodConsent
        period={period.toLowerCase()}
        periodPartnersConsent={periodPartnersConsent}
      />
    </View>
  );
};

export default withLocalization(ProgressChart);
