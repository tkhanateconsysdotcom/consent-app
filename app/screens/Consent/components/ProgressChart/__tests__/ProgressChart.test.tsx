import React from "react";
import { create } from "react-test-renderer";

import ProgressChart from "../ProgressChart";

jest.unmock("../ProgressChart");

const props = {
  totalPartners: 900,
  foodPartners: 200,
  transportPartners: 600,
  otherPartners: 100,
  monthlyPartnersConsent: 1000,
};

it("ProgressChart renders correctly", () => {
  const tree = create(<ProgressChart {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("ProgressChart should not render if monthlyPartnersConsent equals 0", () => {
  const tree = create(
    <ProgressChart {...props} monthlyPartnersConsent={0} />
  ).toJSON();
  expect(tree).toBeNull();
});
