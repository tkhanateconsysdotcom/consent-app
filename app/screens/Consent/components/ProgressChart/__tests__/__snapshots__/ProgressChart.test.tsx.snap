// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressChart renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#DCF0E5",
      "borderRadius": 14,
      "borderWidth": 2,
      "flex": 1,
      "marginHorizontal": 0,
      "marginVertical": 24,
      "paddingBottom": 14,
      "paddingHorizontal": 14,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginVertical": 4,
      }
    }
  >
    <Text.H3
      style={
        Object {
          "textTransform": "capitalize",
        }
      }
    >
      2021
    </Text.H3>
  </View>
  <Chart
    foodPartnersPercentage={0.016666666666666666}
    totalPartnersPercentage={0.075}
    transportPartnersPercentage={0.05}
  />
  <Legend
    foodPartners={200}
    otherPartners={100}
    totalPartners={900}
    transportPartners={600}
  />
  <PeriodConsent
    period="2021"
    periodPartnersConsent={12000}
  />
</View>
`;
